### ISSUE TITLE
Incorrect parsing of default values when GQL context variables are used

### ISSUE THREAD

**Issue by ayush3797** [Azure/data-api-builder#2079]

**Comment by seantleonard** [comment]
The only way I got the request to succeed was to set the variable's content to:
```json
{
    "item": {
        "int_types": 1,
        "float_types": 3.1,
        "uuid_types": "e4826b30-1cbb-4ee1-b09e-b76205a77074"
        }
}
```

**Comment by seantleonard** [comment]
Exception stack trace which indicates that HotChocolate detects that only some of the fields of the mutation are provided and attempts to resolve default values for the remainder of the fields. This violates our assumption that nullable fields are left null. As a result, my current hypothesis is that our GraphQL schema creation may have added default values.. 

![Image](https://github.com/Azure/data-api-builder/assets/6414189/5637afb7-88bb-44dd-b08c-48d3e7b8f11b)

**Comment by seantleonard** [comment]
Yes our schema creation added default value directives. But the Types look inaccurate. I see "String" type which aligns with error message from hot chocolate:


![Image](https://github.com/Azure/data-api-builder/assets/6414189/dbb204e5-0bc9-4b0c-b2ed-a4a2525b1919)

**Comment by seantleonard** [comment]
We need to use the code written in #1883 to handle the following default values returned by sql and create a function to remove parenthesis in default values:

```csharp
// Remove parens:
// 1. built_in funtions are surrounded by single parenthesis,i.e. (getdate())
// 2. string will be with parenthesis and single quotes, i.e. ('value')
// 3. int/float will be with double parenthesis, i.e. ((20))
```

The current form of default value looks to always be a string due to the parens. not sure why this doesn't fail requests that don't provide GraphQL parameters and provide field values for mutations directly in requests.

**Comment by seantleonard** [comment]
This change is more involved and would benefit from sufficient time to test and devise an appropriate solution for all database types because all database types utilize default values but their SDKs (SQL Client and mysql/pg equivalent may return default values differently).
